main.py

```diff
@@ -118,22 +113,31 @@ def setup_target_task_training(args, target_tasks, model, strict):
         # Look for target train checkpoints (available only if we're restoring from a run that
         # already finished), then look for training checkpoints.
 
-        best_path = get_best_checkpoint_path(args.run_dir)
+        best_path = get_best_checkpoint_path(args.run_dir, "target_train")
         if best_path:
             load_model_state(
                 model, best_path, args.cuda, task_names_to_avoid_loading, strict=strict
             )
         else:
-            assert_for_log(
-                args.allow_untrained_encoder_parameters, "No best checkpoint found to evaluate."
-            )
-
-            if args.transfer_paradigm == "finetune":
-                # Save model so we have a checkpoint to go back to after each
-                # task-specific finetune.
-                model_state = model.state_dict()
-                model_path = os.path.join(args.run_dir, "model_state_untrained_pre_target_train.th")
-                torch.save(model_state, model_path)
+            if args.do_pretrain == 1:
+                best_pretrain = get_best_checkpoint_path(args.run_dir, "pretrain")
+                if best_pretrain:
+                    load_model_state(
+                        model, best_pretrain, args.cuda, task_names_to_avoid_loading, strict=strict
+                    )
+                else:
+                    assert_for_log(
+                        args.allow_untrained_encoder_parameters, "No best checkpoint found to evaluate."
+                    )
+
+                if args.transfer_paradigm == "finetune":
+                    # Save model so we have a checkpoint to go back to after each
+                    # task-specific finetune.
+                    model_state = model.state_dict()
+                    model_path = os.path.join(
```

Comment:
@W4ngatang by the way get_best_checkpoint is currently structured, if we do pretraining then target task training, then wouldn't we during target training always load and use the best pretrained model for finetuning? 
If so, then why is this untrained_pre_target_train necessary (as opposed to just always loading the best pretrained model before each finetuning target training for each task?

Reply:
Sometimes we don't want to do any pretraining (e.g. we just want to use BERT for a set of target tasks), so we to create a checkpoint to refer back to at the start of each fine-tuning.

Reply:
So I think this should happen outside of the pretrain check, i.e. you create this if you are targe training but not pretraining.